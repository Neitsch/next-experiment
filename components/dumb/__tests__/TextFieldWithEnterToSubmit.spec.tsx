import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import React from "react";

import TextFieldWithEnter from "../TextFieldWithEnterToSubmit";

jest.unmock("../TextFieldWithEnterToSubmit");

describe("TextFieldWithEnterToSubmit", () => {
  it("renders", () => {
    expect(
      shallow(<TextFieldWithEnter onSubmit={jest.fn()} />),
    ).toMatchSnapshot();
  });
  it("forwards props to textfield", () => {
    const fn = jest.fn();
    expect(toJson(shallow(<TextFieldWithEnter onSubmit={fn} />))).not.toEqual(
      toJson(
        shallow(
          <TextFieldWithEnter
            onSubmit={fn}
            textFieldProps={{ additionalProps: "prop data" }}
          />,
        ),
      ),
    );
  });
  it("Changed textcontent", () => {
    const TextFieldRender = shallow(
      <TextFieldWithEnter onSubmit={jest.fn()} />,
    );
    TextFieldRender.setState({ textcontent: "Test" });
    expect(TextFieldRender.props().value).toEqual("Test");
  });
  it("On Change", () => {
    const TextFieldRender = shallow(
      <TextFieldWithEnter onSubmit={jest.fn()} />,
    );
    TextFieldRender.props().onChange({
      currentTarget: { value: "Test Value" },
      preventDefault: jest.fn(),
    });
    expect(TextFieldRender.state("textcontent")).toEqual("Test Value");
  });
  describe("On Key Press", () => {
    it("Not enter pressed", () => {
      const onSubmit = jest.fn();
      const TextFieldRender = shallow(
        <TextFieldWithEnter onSubmit={onSubmit} />,
      );
      TextFieldRender.setState({ textcontent: "Test" });
      TextFieldRender.props().onKeyPress({ charCode: 97 });
      expect(onSubmit).not.toHaveBeenCalled();
      expect(TextFieldRender.state("textcontent")).toEqual("Test");
    });
    it("Enter pressed", () => {
      const onSubmit = jest.fn();
      const TextFieldRender = shallow(
        <TextFieldWithEnter onSubmit={onSubmit} />,
      );
      TextFieldRender.setState({ textcontent: "Test" });
      TextFieldRender.props().onKeyPress({
        charCode: 13,
        preventDefault: jest.fn(),
      });
      expect(onSubmit).toHaveBeenCalledWith("Test");
      expect(TextFieldRender.state("textcontent")).toEqual("Test");
    });
    it("Enter pressed - clear", () => {
      const onSubmit = jest.fn();
      const TextFieldRender = shallow(
        <TextFieldWithEnter onSubmit={onSubmit} doClear />,
      );
      TextFieldRender.setState({ textcontent: "Test" });
      TextFieldRender.props().onKeyPress({
        charCode: 13,
        preventDefault: jest.fn(),
      });
      expect(onSubmit).toHaveBeenCalledWith("Test");
      expect(TextFieldRender.state("textcontent")).toEqual("");
    });
  });
});
