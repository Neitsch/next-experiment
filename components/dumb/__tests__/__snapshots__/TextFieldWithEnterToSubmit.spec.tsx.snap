// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldWithEnterToSubmit renders 1`] = `
<TextField
  onChange={[Function]}
  onKeyPress={[Function]}
  required={false}
  select={false}
  value=""
  variant="standard"
/>
`;
